lib/src/components/NxGlobalSidebar/__tests__/NxGlobalSidebar.test.tsx

```diff
@@ -5,98 +5,200 @@
  */
 import React from 'react';
-import { ShallowWrapper } from 'enzyme';
-import 'jest-enzyme';
-import { faCrow, faBiohazard } from '@fortawesome/free-solid-svg-icons';
-
-import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
 import NxGlobalSidebar, { Props } from '../NxGlobalSidebar';
-import NxButton from '../../NxButton/NxButton';
-import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { faBiohazard, faCrow } from '@fortawesome/free-solid-svg-icons';
+import { rtlRender, runTimers, userEvent } from '../../../__testutils__/rtlUtils';
+import { screen, within } from '@testing-library/react';
+import NxGlobalSidebarNavigation from '../NxGlobalSidebarNavigation';
+import NxGlobalSidebarNavigationLink from '../NxGlobalSidebarNavigationLink';
+import NxGlobalSidebarFooter from '../NxGlobalSidebarFooter';
 
 describe('NxGlobalSidebar', function() {
-  let minimalProps: Props,
-      toggleOpenSpy: () => void,
-      getShallowComponent: (optionalProps?: Partial<Props>) => ShallowWrapper;
-
-  beforeEach(function() {
-    toggleOpenSpy = jest.fn();
-    minimalProps = {
-      isOpen: false,
-      toggleOpenIcon: faCrow,
-      toggleCloseIcon: faBiohazard,
-      onToggleClick: toggleOpenSpy,
-      logoImg: 'logoImg',
-      logoAltText: 'alt text',
-      logoLink: '/ref'
-    };
-    getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxGlobalSidebar, minimalProps);
-  });
+  const minimalProps: Props = {
+    isOpen: false,
+    toggleOpenIcon: faCrow,
+    toggleCloseIcon: faBiohazard,
+    onToggleClick: () => {},
+    logoImg: 'logoImg',
+    logoAltText: 'alt text',
+    logoLink: '/ref'
+  };
+
+  const quickRender = rtlRender<Props>(NxGlobalSidebar, minimalProps);
 
-  it('renders an aside with nx-global-sidebar class', function() {
-    expect(getShallowComponent()).toMatchSelector('aside.nx-global-sidebar');
+  it('renders a top-level element with role="group"', function() {
+    const view = quickRender(),
+        group = view.getByRole('group');
+
+    expect(group).toHaveAccessibleName('global sidebar');
+    expect(group).toBe(view.container.firstElementChild);
   });
 
-  it('adds classes based on isOpen state', function() {
-    expect(getShallowComponent({ isOpen: true })).toMatchSelector('.nx-global-sidebar.open');
-    expect(getShallowComponent({ isOpen: false })).toMatchSelector('.nx-global-sidebar.closed');
+  describe('children', function() {
+    describe('when only the NxGlobalSidebarNavigation is a provided as a child', function () {
+      it('renders an element with role="navigation" within an element with role="complementary"', function() {
+        const children =
+          <NxGlobalSidebarNavigation key="1">
+            <NxGlobalSidebarNavigationLink icon={faCrow} text="testLink" href="testLink"/>
+          </NxGlobalSidebarNavigation>,
+            view = quickRender({ ...minimalProps, children }),
+            group = view.getByRole('group'),
```

Comment:
Since the test above confirms that the top-level element is the `group`, I think you can skip doing `within(group).getByRole('complementary')` and instead just do `view.getByRole('complementary')`

Reply:
> Since the test above confirms that the top-level element is the `group`, I think you can skip doing `within(group).getByRole('complementary')` and instead just do `view.getByRole('complementary')`

Changed here: 91ffc6b835712f1865e7a7c49f391747870d40e1